cpp/src/Slice/Grammar.y

```diff
@@ -1051,32 +1061,39 @@ class_extends
     ContainerPtr cont = unit->currentContainer();
     TypeList types = cont->lookupType(scoped->v);
     $$ = 0;
-    if(!types.empty())
+    if (!types.empty())
     {
         ClassDeclPtr cl = ClassDeclPtr::dynamicCast(types.front());
-        if(!cl)
+        if (!cl)
         {
-            string msg = "`";
-            msg += scoped->v;
-            msg += "' is not a class";
-            unit->error(msg);
+            if (auto alias = TypeAliasPtr::dynamicCast(types.front()))
+            {
+                if (!alias->typeMetadata().empty())
```

Comment:
This check is specifically for parsing:
`class Bar extends FooAlias`
We forbid any metadata to be specified on `FooAlias`, since that would over-complicate inheritance.
So here, we check if any metadata was specified on the alias, and if there was, we raise an error.

So to answer your question: None can appear here! And this check is the reason why it can't.

Reply:
This check is in the wrong spot. 
When you define a typealias for a class, the Parser should disallow any metadata. So when you call typeMetadata() on a typealias for a class, you always get an empty list.

Reply:
I moved the checks so now defining metadata on a type-alias of a class or interface will report the error at type-alias creation, instead of where it's used.